### PR TITLE
docs: add more details and rationale on the `unchallenged_exit` event

### PR DESCRIPTION
**Pull request by pdobacz**
Answers requests to document on our slack

## Overview

Provides details as in title

## Changes

N/A

## Testing

:eyes: :book: :scroll:

### CODE REVIEW

File: docs/api_specs/status_events_specs.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ Indicates that an invalid exit is occurring. It should be challenged.
 ```
```
Comment (InoMurko) [inline docs/api_specs/status_events_specs.md]: @pdobacz this document says `User should exit.`. What it really means is that the watcher should exit with all it's UTXOs? IS that what User means in this case?

File: docs/api_specs/status_events_specs.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ Indicates that an invalid exit is occurring. It should be challenged.
...
```
Comment (InoMurko) [inline docs/api_specs/status_events_specs.md]: the document is saying that `invalid_ife_challenge` is not implemented yet, is this a blocker for mainnet? cc @Pongch

Comment (pdobacz) [reply docs/api_specs/status_events_specs.md]: These are stale comments, everything that here said "not implemented" is implemented. Nice catch!

File: docs/api_specs/status_events_specs.md (hunk 1 above)
Comment (pdobacz) [reply docs/api_specs/status_events_specs.md]: The User here is the user of the Watcher. Watcher is understood as the app, not a role. 

In other words a "user" here is someone who holds value on the child chain, is interested in keeping that safe and uses the Watcher to that end

File: docs/exit_validation.md [hunk 3]
```diff
@@ -51,6 +50,8 @@ This is acceptable as long as the delay doesn't exceed `scheduled finalization t
 
 Conditions 3 and 4 can be represented jointly, by **continuous** (every child block) validation of the exits.
 If any exit is invalid and its finalization is near, then actions listed under 3 and 4 should take place.
+Let's call this joint representation of 3 and 4 `unchallenged_exit`.
```
Comment (InoMurko) [inline docs/exit_validation.md]: When the text says: `watcher must not allow spending and depositing` - is that prohibited on ChildChain as well? Or could a "hacker" exploit that assumption and allow spending/depositing from the watcher?

Comment (InoMurko) [reply docs/exit_validation.md]: When the text says: `(optional) If finalization of an invalid exit occurs, the watcher must prompt an exit without guarantees of recovery of funds` this basically means FIFO? Whoever is first to exit can get _money back_, if you're last, maybe there won't be anything left to exit with from the contract?

Comment (pdobacz) [reply docs/exit_validation.md]: This not allowing has nothing to do with the child chain. The Watcher protects its user, by not allowing certain actions under certain conditions. The child chain cannot possibly protect the user like that: it isn't trusted by the user.

Or, hmmm, maybe I don't get the question here... what would precisely such hacker do?

Comment (InoMurko) [reply docs/exit_validation.md]: I'm just exploring the possibility of a hacker cloning our repository, modifying the code by *allowing* spending/depositing in certain conditions and how that would affect the operators child chain...

Comment (pdobacz) [reply docs/exit_validation.md]: > When the text says: `(optional) If finalization of an invalid exit occurs, the watcher must prompt an exit without guarantees of recovery of funds` this basically means FIFO? 

FIFO is not 100% precise, as the Priority of exiting has richer logic. 

> Whoever is first to exit can get _money back_, if you're last, 

Again, not only "first to exit" but the exit priority counts here. It is impacted both by time exit started and age of utxo, see `tesuji_blockchain....` docs and others.

Having said that

> maybe there won't be anything left to exit with from the contract?

yep that can happen in this particular scenario (which we're avoiding by following protocol/Watching etc.)

Comment (pdobacz) [reply docs/exit_validation.md]: > I'm exploring the possibility of a hacker cloning our repository, modifying the code by _allowing_ spending/depositing in certain conditions and how that would affect the operators child chain...

If someone uses such modified code, anything can happen. In particular one can turn this particular function of the Watcher off and whoever uses that is exposed. Assuming the operator is using unmodified child chain code, it is never affected by such Watcher's operation.

**Comment by InoMurko** [comment]
This was a very good read!!!

**Pull request merged**